modules/core/src/v2/coins/dot.ts

```diff
@@ -96,6 +102,40 @@ export class Dot extends BaseCoin {
     };
   }
 
+  /** @inheritDoc */
+  deriveKeypair(params: DeriveKeypairOptions): KeyPair | undefined {
+    try {
+      if (_.isNil(params.addressDerivationPrv)) {
+        throw new Error('addressDerivationPrv is missing');
+      }
+      const rootKeyPair = new accountLib.Dot.KeyPair({ prv: params.addressDerivationPrv });
+      const derivationPath = `m/0'/0'/0'/${params.index}'`;
+      const derivedKeys = rootKeyPair.deriveHardened(derivationPath);
+      if (_.isNil(derivedKeys?.prv)) {
+        throw new Error('Key derivation failed - missing derived prv key');
+      }
+      const derivedBase58KeyPair = new accountLib.Dot.KeyPair({ prv: derivedKeys.prv });
+      const { prv, pub } = derivedBase58KeyPair.getKeys();
+      const derivedAddress = derivedBase58KeyPair.getAddress();
+      if (!_.isString(prv) || !accountLib.Dot.Utils.default.isValidPrivateKey(prv)) {
+        throw new Error('failed to create valid derived base58 prv key');
+      }
+      if (!_.isString(pub) || !accountLib.Dot.Utils.default.isValidPublicKey(pub)) {
+        throw new Error('failed to create valid derived base58 pub key');
+      }
+      if (!accountLib.Dot.Utils.default.isValidAddress(derivedAddress)) {
+        throw new Error('failed to create valid DOT address from derived base58 keys');
+      }
+      return {
+        prv: '',
```

Comment:
I think you gonna need to return this prv since is needed to sign consolidation transfer for example.

Reply:
ah okay - I was wondering where it was getting used. thanks.